plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java

```diff
@@ -125,10 +127,10 @@ protected void testCreateTableFailure(String expectedExceptionMessage, boolean s
 
         Path metadataDirectory = Path.of(tableLocation, "metadata");
         if (shouldMetadataFileExist) {
-            assertThat(metadataDirectory).as("Metadata file should exist").isDirectoryContaining("glob:**.metadata.json");
+            assertThat(metadataDirectory).as("Metadata file should exist").isDirectoryContaining(METADATA_GLOB);
         }
         else {
-            assertThat(metadataDirectory).as("Metadata file should not exist").isEmptyDirectory();
+            assertThat(metadataDirectory).as("Metadata file should not exist").isDirectoryNotContaining(METADATA_GLOB);
```

Comment:
This is a regression. Why did the test start leaving files in `metadata` directory?

Reply:
https://github.com/trinodb/trino/pull/19188#issuecomment-1740113189

Reply:
@nastra Please restore the original behavior or leave TODO comment.

Reply:
@ebyhr what kind of TODO would you like me to add here. This behavioral change is expected according to the comments in https://github.com/apache/iceberg/pull/8675.
cc @Fokko

Reply:
This was discussed in the community sync, see https://github.com/apache/iceberg/pull/8599. The new behavior is more defensive in cleaning up files in case of a conflict. This might introduce lingering files but allow it to recover to a certain version, or be able to backtrack what happened.

Reply:
Then, let's leave a code comment explaining the context.

Reply:
added a comment and mentioned the Iceberg PR